hikari/impl/interaction_server.py

```diff
@@ -441,7 +459,15 @@ async def on_interaction(self, body: bytes, signature: bytes, timestamp: bytes)
         if listener := self._listeners.get(type(interaction)):
             _LOGGER.debug("Dispatching interaction %s", interaction.id)
             try:
-                result = await listener(interaction)
+                call = listener(interaction)
+
+                if inspect.isasyncgen(call):
+                    result = await call.__anext__()
+                    asyncio.create_task(_consume_generator_handler(call))
```

Comment:
as always you'd need to save these tasks somewhere to make sure they aren't cancelled by GC

Reply:
Due to the way asyncio is built, the task will never get gc'ed, but should probably be keeping it somehow anyways. Forgot about that, thanks for pointing it out!

Reply:
> Due to the way asyncio is built, the task will never get gc'ed

That isn't guaranteed behaviour but I do remember having issues with production runs randomly cancelling long running tasks before.

The documented behaviour is that 
> https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

"Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks."

Reply:
Resolved :)